src/envoy/utils/mixer_control.cc

```diff
@@ -100,5 +102,67 @@ Grpc::AsyncClientFactoryPtr GrpcClientFactoryForCluster(
   return std::make_unique<EnvoyGrpcAsyncClientFactory>(cm, service);
 }
 
+/** example node
+   "node": {
+     "id":
+"sidecar~10.36.0.15~fortioclient-84469dc8d7-jbbxt.service-graph~service-graph.svc.cluster.local",
+     "cluster": "fortioclient",
+     "metadata": {
+      "ISTIO_VERSION": "1.0.1",
+      "POD_NAME": "fortioclient-84469dc8d7-jbbxt",
+      "istio": "sidecar",
+      "INTERCEPTION_MODE": "REDIRECT",
+      "ISTIO_PROXY_VERSION": "1.0.0",
+      "ISTIO_PROXY_SHA": "istio-proxy:2656f34080413d3aec444aa659cc78057508c57b"
+     },
+     "build_version": "0/1.8.0-dev//RELEASE"
+    },
+
+    ==> uid: kubernetes://fortioclient-84469dc8d7-jbbxt.service-graph
+    reporter == uid
+    namespace
+    IP_Address only for inbound.
+**/
+const LocalAttributes *GenerateLocalAttributes(
+    const LocalInfo::LocalInfo &local_info) {
+  auto parts = StringUtil::splitToken(local_info.node().id(), "~");
+  if (parts.size() < 3) {
+    GOOGLE_LOG(ERROR)
+        << "GenerateLocalAttributes error len(node.id.split(~))<3: "
+        << local_info.node().id();
+    return nullptr;
+  }
+
+  auto longname = std::string(parts[2].begin(), parts[2].end());
+  auto names = StringUtil::splitToken(longname, ".");
+  if (names.size() < 2) {
+    GOOGLE_LOG(ERROR)
+        << "GenerateLocalAttributes error len(split(longname, '.')) < 3: "
+        << longname;
+    return nullptr;
+  }
+
+  std::string ns = std::string(names[1].begin(), names[1].end());
+  std::string uid = "kubernetes://" + longname;
+
+  ::istio::mixer::v1::Attributes ib;
+  auto &inbound = (*ib.mutable_attributes());
```

Comment:
Can we use AttributeBuilder defined here: https://github.com/istio/proxy/blob/master/include/istio/utils/attributes_builder.h